src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java

```diff
@@ -1181,12 +1188,20 @@ public Location getCompassTarget()
 	{
 		return this.compassTarget;
 	}
+	/**
+	 * Sets the {@link InetSocketAddress} returned by {@link #getAddress}.
+	 *
+	 * @param address The address to set.
+	 */
+	public void setAddress(@Nullable InetSocketAddress address)
+	{
+		this.address = address;
+	}
 
 	@Override
-	public InetSocketAddress getAddress()
+	public @Nullable InetSocketAddress getAddress()
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		return (getServer().getPlayer(getUniqueId()) == null) ? null : address;
```

Comment:
I reckon it would be good to have a method for this, e.g. `PlayerMock#isOnline()`
We could also re-use that method in the context of #352 then too

```suggestion
		return isOnline() ? null : address;
```